pkg/services/live/liveplugin/plugin.go

```diff
@@ -59,6 +59,27 @@ func (p *NumLocalSubscribersGetter) GetNumLocalSubscribers(channelID string) (in
 	return p.node.Hub().NumSubscribers(channelID), nil
 }
 
+type ChannelUnsubscriber struct {
+	node *centrifuge.Node
+}
+
+func NewChannelUnsubscriber(node *centrifuge.Node) *ChannelUnsubscriber {
+	return &ChannelUnsubscriber{node: node}
+}
+
+func (c *ChannelUnsubscriber) Unsubscribe(channelID string) error {
+	p, err := c.node.Presence(channelID)
+	if err != nil {
+		return err
+	}
+	for _, v := range p.Presence {
+		if err = c.node.Unsubscribe(v.UserID, channelID); err != nil {
```

Comment:
This will unsubscribe a user from a channel on all nodes, even if channel stream on the other node is still working. Ideally we need to have https://github.com/grafana/grafana/pull/43145 to not think about things like this at all. It's also possible to unsubscribe local subscribers only - but this is a step away from non-hacky implementation :(